simvue_fds/connector.py

```diff
@@ -226,6 +228,75 @@ def _find_fds_executable(self):
 
         return fds_bin
 
+    def _map_line_var_coords(self):
+        """Map DEVC line variables to their coordinates."""
+        _labels = ("x", "y", "z")
+        # Loop through input dict and find all DEVC devices
+        for key, devc in self._input_dict.items():
+            if "devc" in key:
+                # If devc device does not have an ID, cannot add to mapping, so ignore
+                if not (_devc_id := devc.get("id")):
+                    continue
+                # Figure out which axes it varies along
+                _devc_coords = devc.get("xb") or devc.get("xbp")
+                if not _devc_coords:
+                    continue
+
+                _devc_coords_change = [
+                    _devc_coords[1] - _devc_coords[0],
+                    _devc_coords[3] - _devc_coords[2],
+                    _devc_coords[5] - _devc_coords[4],
+                ]
+
+                _devc_coords_idx = [
+                    idx for idx, val in enumerate(_devc_coords_change) if abs(val) > 0
+                ]
+
+                if len(_devc_coords_idx) == 0:
+                    # Indicates a point DEVC device using XB or XBP definition - dont add to mapping, just continue
+                    continue
+
+                elif len(_devc_coords_idx) > 1:
+                    # Not varying in 1D, currently not supported # TODO
+                    logger.warning(
+                        f"Multi-dimensional DEVC device '{_devc_id}' detected - these are currently not supported and will be ignored."
+                    )
+                    continue
+                else:
```

Comment:
I think `else` is not needed here as length cannot be negative and you cover `>1` and `== 0` already.